Data/EitherR.hs

```diff
@@ -79,6 +85,14 @@ import qualified Control.Monad.Trans.Except
 -}
 newtype EitherR r e = EitherR { runEitherR :: Either e r }
 
+appendEitherR :: EitherR r e -> EitherR r e -> EitherR r e
+appendEitherR (EitherR x) (EitherR y) = EitherR $ flipEither $ flipEither x <> flipEither y
```

Comment:
You will need to go with the solution listed to conditionally depend on `semigroups`:

```
if !impl(ghc >= 8.0)
  build-depends: semigroups == 0.18.*
```

... instead of conditionally importing `Data.Semigroup`.  Otherwise, `appendEitherR` won't compile on older versions of `base`